src/pages/workspaces/workspace/analysis/modals/ComputeModal.js

```diff
@@ -1798,7 +1796,8 @@ export const ComputeModalBase = ({
                   'Please delete the existing disk before selecting a new type.'
                 ],
                 side: 'bottom'
-              }, [renderPersistentDiskType(idDiskType)]) : renderPersistentDiskType(idDiskType),
+              // eslint-disable-next-line react-hooks/rules-of-hooks
+              }, [renderPersistentDiskType(useUniqueId())]) : renderPersistentDiskType(useUniqueId()),
```

Comment:
There is no reason to pass the unique ID as an argument to `renderPersistentDiskType`. Change `renderPersistentDiskType` itself to not have any input arguments, and instead create the ID in its own method.

Reply:
also, in general, resorting to // eslint-disable-next-line, especially something like rules-of-hooks, should be a giant red flag... hooks should ONLY be called at the top of a functional component so that the exact order and number of hooks calls is consistent between render calls.  But as Christina suggests, this pattern can be avoided completely by moving the uniqueID handling to be inside renderPersistentDiskType which should be a simple React Functional component instead of a "render helper".  Actually it will NEED to be a funcitonal component to leverage the suggested preferred useUniqueId hook from IDContainer (there is an outstanding ticket to clean up the dual options for uniqueId later)